compiler/src/main/java/com/squareup/anvil/compiler/ModuleMergerIr.kt

```diff
@@ -253,30 +259,20 @@ internal class ModuleMergerIr(
       .distinct()
       .map { it.clazz.owner }
 
-    val annotationConstructorCall = IrConstructorCallImpl
-      .fromSymbolOwner(
-        startOffset = UNDEFINED_OFFSET,
-        endOffset = UNDEFINED_OFFSET,
-        type = pluginContext.requireReferenceClass(daggerAnnotationFqName).defaultType,
-        constructorSymbol = pluginContext
-          .referenceConstructors(daggerAnnotationFqName.classIdBestGuess())
-          .single { it.owner.isPrimary }
-      )
+    val annotationConstructorCall = irCallConstructor(
+      callee = pluginContext
+        .referenceConstructors(daggerAnnotationFqName.classIdBestGuess())
+        .single { it.owner.isPrimary },
+      typeArguments = emptyList()
+    )
       .apply {
         putValueArgument(
           index = 0,
-          valueArgument = IrVarargImpl(
-            startOffset = UNDEFINED_OFFSET,
-            endOffset = UNDEFINED_OFFSET,
-            type = pluginContext.symbols.array.defaultType,
-            varargElementType = IrDynamicTypeImpl(null, emptyList(), INVARIANT),
-            elements = contributedModules
+          valueArgument = irVararg(
+            elementType = pluginContext.irBuiltIns.kClassClass.starProjectedType,
```

Comment:
This is a change from before, but I think this is more correct (`Array<KClass<*>>`) as `dynamic` is only really intended for JS targets iirc

Reply:
What do you mean by this being a change from before?

Reply:
Before it was surprisingly the kotlin dynamic type, which I suspect was for lack of a better star alternative: https://kotlinlang.org/docs/dynamic-type.html

Reply:
`The dynamic type is not supported in code targeting the JVM.` - mildly concerning :D, apparently it works though 😆. Good catch.